src/utilities/alfalfa/AlfalfaPoint.cpp

```diff
@@ -0,0 +1,137 @@
+namespace alfalfa {
+  namespace detail {
+    AlfalfaPoint_Impl::AlfalfaPoint_Impl() {}
+  }  // namespace detail
+
+  AlfalfaPoint::AlfalfaPoint() : m_impl(std::shared_ptr<detail::AlfalfaPoint_Impl>(new detail::AlfalfaPoint_Impl())) {}
+
+  AlfalfaPoint::AlfalfaPoint(const std::string& display_name) : m_impl(std::shared_ptr<detail::AlfalfaPoint_Impl>(new detail::AlfalfaPoint_Impl())) {
+    setDisplayName(display_name);
+  }
+
+  void AlfalfaPoint::setInput(const AlfalfaComponent& component) {
+    if (component.canInput()) {
+      m_impl->m_input = component;
+    } else {
+      throw std::runtime_error(fmt::format("Component of type: {} cannot be used as an input.", component.type));
+    }
+  }
+
+  boost::optional<AlfalfaComponent> AlfalfaPoint::input() const {
+    return m_impl->m_input;
+  }
+
+  void AlfalfaPoint::setOutput(const AlfalfaComponent& component) {
+    if (component.canOutput()) {
+      m_impl->m_output = component;
+    } else {
+      throw std::runtime_error(fmt::format("Component of type: {} cannot be used as an output.", component.type));
+    }
+  }
+
+  boost::optional<AlfalfaComponent> AlfalfaPoint::output() const {
+    return m_impl->m_output;
+  }
+
+  void AlfalfaPoint::setUnits(const std::string& units) {
+    m_impl->m_units = units;
+  }
+
+  boost::optional<std::string> AlfalfaPoint::units() const {
+    return m_impl->m_units;
+  }
+
+  boost::optional<std::string> AlfalfaPoint::id() const {
+    boost::optional<std::string> result = m_impl->m_id;
+    if (!result.is_initialized()) {
+      std::string id = toIdString(displayName());
+      if (isValidId(id)) {
+        result = id;
+      } else {
+        LOG(Warn, fmt::format("Display name '{}' does not produce a valid point ID. Manually set a valid ID or export will fail.", displayName()));
+      }
+    }
+    return result;
+  }
+
+  void AlfalfaPoint::setId(const std::string& id) {
+    if (isValidId(id)) {
+      m_impl->m_id = id;
+    } else {
+      throw std::runtime_error(ID_VALID_CHARS_MSG);
+    }
+  }
+
+  std::string AlfalfaPoint::displayName() const {
+    return m_impl->m_display_name;
+  }
+
+  void AlfalfaPoint::setDisplayName(const std::string& display_name) {
+    if (display_name.size() == 0) {
+      throw std::runtime_error("Display name must have non-zero length");
+    }
+    m_impl->m_display_name = display_name;
+    if (!m_impl->m_id.is_initialized()) {
+      std::string id = toIdString(display_name);
+      if (!isValidId(id)) {
+        LOG(Warn, fmt::format("Display name '{}' does not produce a valid point ID. Manually set a valid ID or export will fail.", display_name));
+      }
+    }
+  }
+
+  bool AlfalfaPoint::optional() const {
+    return m_impl->m_optional;
+  }
+
+  void AlfalfaPoint::setOptional(bool optional) {
+    m_impl->m_optional = optional;
+  }
+
+  Json::Value AlfalfaPoint::toJSON() const {
+    Json::Value point;
+
+    if (id().is_initialized()) {
+      point["id"] = id().get();
```

Comment:
Two calls to id() here. Same comment applies to all other occurences of this pattern below.

```suggestion
    if (auto id_ = id().is_initialized()) {
      point["id"] = *id_;
```

notes:
* you can also `std::move(*id_)` if relevant
* .get() will internally check if is_initialized to handle failure, which you're already doing. Dereferencing the optional does not so it's faster.